### PR TITLE
p005-ReLU-Activation.rs changes

### PR DESCRIPTION
**Pull request by TheMartian117**
The first contributer, SLASHLogin didn't really make a function Activation_ReLU which took an array and outputted an array, so I decided thats what I'd do. SLASHLogin deserves full credit for the code for the creation of the dataset.

### CODE REVIEW

**Comment by SLASHLogin** [comment]
I think that you should just comment my PR instead of making new one. It's easier to maintain repo and approve PRs.

File: Rust/p005-ReLU-Activation.rs [hunk 1]
```diff
@@ -0,0 +1,115 @@
+#[macro_use]
+
+extern crate ndarray;
+extern crate nalgebra;
```
Comment (SLASHLogin) [inline Rust/p005-ReLU-Activation.rs:4]: You're not even using nalgebra in this code

File: Rust/p005-ReLU-Activation.rs [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+use std::f64;
+use std::convert::From;
+
+struct Layer_Dense {
+	weights: Array2<f64>,
+	biases: Array2<f64>,
+	output: Option<Array2<f64>>,
+}
+impl Layer_Dense {
+ 	fn init(n_inputs: usize, n_neurons: usize) -> Self {
```
Comment (SLASHLogin) [inline Rust/p005-ReLU-Activation.rs:20]: Rust functions returning self, initializing struct should be named new. It's a python name convention to call those functions init

File: Rust/p005-ReLU-Activation.rs [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+
+#[derive(Debug)]
+struct Activation_ReLU {
+	output: Vec<f64>
+}
+
+impl Activation_ReLU {
+	fn init() -> Self {
+		let output = vec![0., 0., 0., 0., 0.];
```
Comment (SLASHLogin) [inline Rust/p005-ReLU-Activation.rs:84]: There's no need to create empty array just to overwrite it's content. Rust optimizer of course should omit this part, but still it's something what's not necessary.

File: Rust/p005-ReLU-Activation.rs [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+		Activation_ReLU{
+			output
+		}
+	}
+
+
+	fn ActivationReLU(&mut self, inputs: Array2<f64>){
+		self.output = inputs.iter().map(|x| x.max(0.0)).collect::<Vec<f64>>();
```
Comment (SLASHLogin) [inline Rust/p005-ReLU-Activation.rs:92]: This function looks good and it's something what I should have written in my PR. Feel free to commit this to the first PR

**Comment by daniel-kukiela** [comment]
@SLASHLogin What do you think about this solution? Should I accept this or your submission?

**Comment by SLASHLogin** [comment]
@daniel-kukiela I think changes proposed in this PR are needless and some parts might cause confusion. I'm open to propositions to change my code if there's anything what might need rewriting.

**Comment by TheMartian117** [comment]
@daniel-kukiela i feel like my solution implements what's needed as simply as possible as @SLASHLogin didn't really create an actual Activation_ReLU function. You can commit which one you think is better though :)

**Comment by SLASHLogin** [comment]
@TheMartian117, you just moved mapping of this array to the function, which I was doing anyway but in main fn. I've said that this way is better and I've told you that you can make PR to my fork and I would add it. But there was no response from you, so I added it myself after Daniel's question

**Comment by daniel-kukiela** [comment]
@TheMartian117 I'm going to accept @SLASHLogin 's solution as he made his submission before you made yours. Feel free to create PRs with modifications if you feel like there still should be some (following @SLASHLogin 's suggestion that you can make them).

**Pull request closed without merging**